### ISSUE TITLE
Request: Add the 'isMoveToNextResult' argument to the QueryMultipleExtractor (Extract and Scalar) methods

### ISSUE THREAD

**Issue by mikependon** [mikependon/RepoDB#591]
As requested by the community, we need to add an additional argument to the `Extract` and `Scalar` method of the [QueryMultipleExtractor](https://repodb.net/class/querymultipleextractor) class.

The call below is moving to the next result after calling the `Extract`.

```csharp
using (var connection = new SqlConnection(connectionString))
{
	var commandText = "SELECT Id, Name FROM [dbo].[Customer] WHERE Id = @CustomerId; SELECT * FROM [dbo].[Order] WHERE CustomerId = @CustomerId;";

	using (var extractor = connection.ExecuteQueryMultiple(commandText, new { CustomerId = 10045 }))
	{
		var customer = extractor.Extract<Customer>().FirstOrDefault(); // <-- This does not moves to the next result
		var orders = extractor.Extract<Order>().AsList(); // <-- In the 2nd position
		
		// Do the stuffs for the 'customer' and 'orders' here
	}
}
```

By passing a value of `false` to the argument `isMoveToNextResult`, then the data reader will not move the pointer.

```csharp
using (var connection = new SqlConnection(connectionString))
{
	var commandText = "SELECT Id, Name FROM [dbo].[Customer] WHERE Id = @CustomerId; SELECT * FROM [dbo].[Order] WHERE CustomerId = @CustomerId;";

	using (var extractor = connection.ExecuteQueryMultiple(commandText, new { CustomerId = 10045 }))
	{
		var customer = extractor.Extract<Customer>(false).FirstOrDefault(); // <-- This moves to the next result
		extractor.NextResult(); // <-- Manually moving to the next result
		var orders = extractor.Extract<Order>().AsList(); // <-- In the 2nd position
		
		// Do the stuffs for the 'customer' and 'orders' here
	}
}
```

The same scenario is the same for the `Scalar` method, and also to their corresponding `Async` methods. FYI: @cleverguy25

**Comment by mikependon** [comment]
This is now available at [RepoDb v1.12.0](https://www.nuget.org/packages/RepoDb/1.12.0).

**Comment by mikependon** [comment]
This has been fixed and is now available at [RepoDb v1.12.3](https://www.nuget.org/packages/RepoDb/1.12.3). Closing this incident ticket now.
